templates/_helpers.tpl

```diff
@@ -156,7 +156,11 @@ Set's the replica count based on the different modes configured by user
   {{ if eq .mode "standalone" }}
     {{- default 1 -}}
   {{ else if eq .mode "ha" }}
-    {{- .Values.server.ha.replicas | default 3 -}}
+    {{- if kindIs "invalid" .Values.server.ha.replicas -}}
```

Comment:
TIL about `kindIs`. I couldn't find any documentation for "invalid" though. Do you have any references you could share? I'm wondering if it might be better to switch the logic though, something like:

```
{{- if kindIs "int" .Values.server.ha.replicas -}}
  {{- .Values.server.ha.replicas -}}
{{- else -}}
  {{- 3 -}}
{{- end -}}
```

Although I haven't thought very deeply about it yet. Do you think that would also work?

Reply:
The source code here just uses Go's `reflect` lib: https://github.com/Masterminds/sprig/blob/master/reflect.go 

I believe both ways will work, but I like your approach better as it seems more intuitive. I'll adapt my PR tomorrow.

Reply:
Based on [this](https://github.com/helm/helm/issues/12084) and some manual fiddling, I changed it to check for `float64`.

Reply:
Thanks for doing the research, that looks good to me! 👍

Reply:
On the tests, are you referring to the vault-helm-test image? We don't actually host that internally, we just build it on demand and it's documented here: https://github.com/hashicorp/vault-helm/blob/main/CONTRIBUTING.md#test-using-docker-container. But that is pretty easy to miss, so apologies if that caused confusion.